### DISCUSSION TITLE
Adding attribute to PPS object from XY data

### DISCUSSION THREAD

**Discussion by gguryan** [wschwanghart/topotoolbox#52] [Q&A] [answered]
Hi,

I'm looking at the spatial distribution of boulders in channels in my study area. I have a table with x-coordinates, y-coordinates, and c-axis lengths for each boulder. I've used this table to create a PPS object where each point represents a boulder.

My question is: Is it possible to attach the boulder size data to each point when creating a PPS object? 

I thought about creating a node attribute list of the boulder lengths, but since some of my boulders are clustered pretty closely together, there's not necessarily a single boulder at each node (and some nodes don't have any boulders).

My ultimate goal is to calculate the cumulative sum of boulder lengths over different segments of the channel. For context, I only have boulders mapped along the trunk channel of each watershed, not the entire network. 

Do you have any suggestions on how I could approach this?

**Comment by wschwanghart** [comment, marked as answer]
Hi,

unfortunately, marks have so far not gotten much attention. Maybe this should be something to look into the future. In fact, PPS was designed to also hold marks and covariates.
Still, for your application, there could be a work-around:

```
DEM = GRIDobj('srtm_bigtujunga30m_utm11.tif');
FD  = FLOWobj(DEM);
S = STREAMobj(FD,'minarea',1000);
S = trunk(klargestconncomps(S,1));

% Bould locations
IX = randlocs(S,100);

% Caxis length
ca = random(makedist('Exponential','mu',3),100,1);

% Channel segments
label = labelreach(S,"seglength",2000);

% PPS
P = PPS(S,"PP",IX,"z",DEM);

% Sum of Caxis length in each channel segment. You could also calculate the
% median, mean, ... using the anonymous function.
sca = accumarray(label(P.PP),ca,[max(label) 1],@sum);

% Map sum of Caxis length back to a new node-attribute list
nalca = sca(label);

% and plot
plotc(S,nalca)
```

I hope this is what you were asking for. 
Best, Wolfgang

**Comment by gguryan** [comment]
This is exactly what I needed, thank you! -Grace
